pkg/localstorev2/internal/pinning/pinning.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pinstore
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/google/uuid"
+)
+
+const (
+	// size of the UUID generated by the pinstore
+	uuidSize = 36
+)
+
+var (
+	// errInvalidPinCollectionAddr is returned when trying to marshal a pinCollectionItem
+	// with a zero address
+	errInvalidPinCollectionAddr = errors.New("marshal pinCollectionItem: address is zero")
+	// errInvalidPinCollectionUUID is returned when trying to marshal a pinCollectionItem
+	// with an empty UUID
+	errInvalidPinCollectionUUID = errors.New("marshal pinCollectionItem: UUID is zero")
+	// errInvalidPinCollectionSize is returned when trying to unmarshal a buffer of
+	// incorrect size
+	errInvalidPinCollectionSize = errors.New("unmarshal pinCollectionItem: invalid size")
+)
+
+// batchSize used for deletion
+var batchSize = 100
+
+// CollectionStat is used to store some basic stats about the pinning collection
+type CollectionStat struct {
+	Total           uint64
+	DupInCollection uint64
+}
+
+// pinCollectionSize represents the size of the pinCollectionItem
+const pinCollectionItemSize = swarm.HashSize + uuidSize + 8 + 8
+
+// pinCollectionItem is the index used to describe a pinning collection. The Addr
+// is the root reference of the collection and UUID is a unique UUID for this collection.
+type pinCollectionItem struct {
+	Addr swarm.Address
+	UUID string
```

Comment:
Is there any other reason, then checking the length, why `UUID` is defined as `string`? If you define it as `[]byte` it'll take less space and you can check its length for `0`.